hbase-server/src/main/java/org/apache/hadoop/hbase/master/procedure/ServerRemoteProcedure.java

```diff
@@ -113,17 +117,20 @@ protected synchronized void completionCleanup(MasterProcedureEnv env) {
   @Override
   public synchronized void remoteCallFailed(MasterProcedureEnv env, ServerName serverName,
     IOException exception) {
+    state = MasterProcedureProtos.ServerRemoteProcedureState.SERVER_REMOTE_PROCEDURE_DISPATCH_FAIL;
     remoteOperationDone(env, exception);
   }
 
   @Override
   public synchronized void remoteOperationCompleted(MasterProcedureEnv env) {
+    state = MasterProcedureProtos.ServerRemoteProcedureState.SERVER_REMOTE_PROCEDURE_REPORT_SUCCEED;
     remoteOperationDone(env, null);
   }
 
   @Override
   public synchronized void remoteOperationFailed(MasterProcedureEnv env,
     RemoteProcedureException error) {
+    state = MasterProcedureProtos.ServerRemoteProcedureState.SERVER_REMOTE_PROCEDURE_SERVER_CRASH;
```

Comment:
I do not think this should be server crash? The remote region server could also report the failure?

Reply:
For this, I rely on the [documentation comments](https://github.com/apache/hbase/pull/5816/files#diff-474e3ef302fa94ddd545c9170a788e9ece0eb4a9ea3ca9443c288771a731a2dfL66). 
```
 If the targetServer crashed but this
 * procedure has no response, than dispatcher will call remoteOperationFailed()
```

Reply:
I go through the code and find that this is not the case. Let me change it in that scenario.